modules/reactor/src/meshgenerators/FlexiblePatternGenerator.C

```diff
@@ -553,9 +556,25 @@ FlexiblePatternGenerator::FlexiblePatternGenerator(const InputParameters & param
 std::unique_ptr<MeshBase>
 FlexiblePatternGenerator::generate()
 {
-  if (isParamValid("external_boundary_id"))
-    MooseMesh::changeBoundaryId(
-        **_build_mesh, OUTER_SIDESET_ID, getParam<boundary_id_type>("external_boundary_id"), false);
+  const auto external_boundary_id =
+      isParamValid("external_boundary_id") ? getParam<boundary_id_type>("external_boundary_id") : 0;
+  const auto external_boundary_name =
```

Comment:
Is it possible that `external_boundary_name` is already assigned to a boundary id that differs from `external_boundary_id`? If so, we need to check here.

Reply:
Added a check for this